### PR TITLE
coveralls.yml: Improve coverage mapping and remove 1.1.1

### PR DESCRIPTION
**Pull request by t8m**
Enable all things that do not require additional external libraries being installed.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
This will get the coverage up :)

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

File: .github/workflows/coveralls.yml [hunk 1]
```diff
@@ -25,16 +25,14 @@ jobs:
       matrix:
         branches: [
           {
-            branch: OpenSSL_1_1_1-stable
-          }, {
             branch: openssl-3.1,
-            extra_config: enable-fips
+            extra_config: no-afalgeng enable-fips enable-ssl-trace enable-trace enable-zlib enable-rc5 enable-md2 enable-weak-ssl-ciphers enable-ec_nistp_64_gcc_128
```
Comment (mattcaswell) [inline .github/workflows/coveralls.yml:29]: Do we need to do anything additional here to support the zlib dependencies?

**Review by mattcaswell — APPROVED** [review]

File: .github/workflows/coveralls.yml (hunk 1 above)
Comment (t8m) [reply .github/workflows/coveralls.yml:29]: I believe not. We have enable-zlib with other CI jobs without anything additional installed.

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**